### DISCUSSION TITLE
When to delete_fluid_player after fluid_player_join ?

### DISCUSSION THREAD

**Discussion by be1** [FluidSynth/fluidsynth#1281] [Q&A] [answered]
From the main GUI thread of my application, when the _Play_ button is pressed, I start a thread that launches `fluid_player_play`, then waits with `fluid_layer_join` and finally calls `delete_fluid_player` on the main thread after the player thread has returned. I can interrupt playing by calling `fluid_player_stop` from the main thread. Unfortunately in this case, a crash appears sometimes on `fluid_synth_write_*`. The documentation warns* it is unsafe to `delete_fluid_player` after a return of `fluid_player_join`. But when can I delete it then ? I cannot keep the fluid_player alive since I must recreate a fluid_player to reload the (modified) MIDI file on restarting playback, and the API has only `fluid_player_add` to manage MIDI files. Any hint ?

[*] https://www.fluidsynth.org/api/group__midi__player.html#ga9e655b26be62b17c59319a287ef382f3

**Comment by derselbst** [comment, marked as answer]
> The documentation warns* it is unsafe to delete_fluid_player after a return of fluid_player_join. But when can I delete it then ? 

You _can_ delete the player after joining. But additionally, you need to ensure that no other thread concurrently calls any of fluidsynth's rendering functions.
